pending/json-api.pod

```diff
@@ -0,0 +1,290 @@
+=head1 Cookiecuttin' data with JSON:API
+
+A web service providing access to a datastore, that's simple. Even moreso with
+Dancer2. Like, let's say you're on L<Glugg's team|LINK TO FIRST ARTICLE>  and
+maintain the North Pole Wishlist Database service. Then your endpoint might
+look something like, say,
+
+    get '/child/:child_name' =>  sub {
+
+        # database magic replaced by hardcoded
+        # data for the example's sake
+
+        to_json {
+            name => 'cromedome',
+            naughtoscale => 8,
+            wishlist => [
+                'Cessna Skyhawk', 'Verg articulated figure',
+            ],
+        }
+    };
+
+As advertised; simple.
+
+Except...
+
+Except those endpoints never stay simple for long, do they? Those gifts in the
+wishlist? They are probably objects of their own right in the datastore, and
+it'd be nice to be able to follow a thread to get them, somehow. 
+
+Or that same gift list might get too long in the case of greedy kids, so in
+some cases -- for bandwidth's sake -- we
+might want to get the data minus the gifts. 
+
+And if we had
+a list of children, maybe we would like to sort them my name, or by their
+naughtoscale score. 
+
+And what about pagination? 
+
+I'll stop here. But yeah. Those endpoints, they never stay simple for long.
+
+=head2 Doing the same thing over and over again, and expecting the same result
+
+The thing is, all those operations on the data listed in the previous section aren't
+weird feature requests. In fact, it's more surpring to have 
+an API that B<doesn't> get to implement them in one way or
+another. And yet we tend to reinvent the way they are implemented 
+each time they pop up. And that's kind of silly.
+
+Enters L<JSON:API|https://jsonapi.org/>, which is a set of specs that
+prescribes a standard way to serialize database-like data, and how to interact
+with it via a REST-like interface. 
+
+=head2 So what does it look like?
+
+An example will speak volume here. Let's take the data returned by our
+original endpoint.
+
+    {
+        name => 'cromedome',
+        naughtoscale => 8,
+        wishlist => [
+            'Cessna Skyhawk', 'Verg articulated figure',
+        ],
+    }
+
+That same data, when covered in
+JSON:API sauce, could look like this:
+
+    {
+        jsonapi => { version => '1.0 '},
+        links => {
+            self => '/child/cromedome',
+        },
+        data => {
+            id => 'child-83626812-b',
+            type => 'child',
+            attributes => {
+                naughtoscale => 8,
+                name => 'cromedome',
+            }, 
+            relationships=> {
+                gifts => {
+                   links => {
+                        self => '/child/cromedone/wishlist',
```

Comment:
`s/cromedone/cromedome/; # :)`

Reply:
oh my lord, what a typo! That's now totally your end-of-week-can-the-madness-please-stop nickname. :laughing: